packages/r/rustlib/xmake.lua

```diff
@@ -0,0 +1,72 @@
+package("rustlib")
+    set_kind("template")
+    set_description("Template to help with Rust libraries in xrepo")
+ 
+    on_load(function (package)
+        local toolchainconfigs = {}
+        toolchainconfigs.target_plat = package:plat()
+        toolchainconfigs.target_arch = package:arch()
+
+        if package:is_plat("cross") then
+            -- detect cross configuration from the compiler, if possible
+            local compiler, toolname = package:tool("cc")
+            if toolname == "clang" or toolname == "gcc" then
+                local outdata, errdata = os.iorunv(compiler, {"-v"})
```

Comment:
we should not call os.iorunv in on_load

Reply:
Unfortunately that's the only way I found to extract the compiler target (which is a useful information for cross platform)

Reply:
I took a look at the way xmake handles tools, I was thinking of making a pull request to add the support of an `:info()` method, or maybe a complete `:target()` or `:target_abi()`, but I think this is something you probably want to think a bit before. Do you want me to open a feature request?

Reply:
You can submit a feature request first. I haven't figured out whether it is needed and how to design the API, so it may not be added for the time being.